### PR TITLE
Add tests for SyncAggregate with no participants and all zero signature

### PR DESCRIPTION
**Pull request by ajsutton**
Adds a test for `SyncAggregate` processing where there are no participants and the signature is all zeros instead of the infinite point.
Also a test for the infinite signature when there are participants.

### CODE REVIEW

File: tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py [hunk 1]
```diff
@@ -113,6 +113,36 @@ def test_invalid_signature_missing_participant(spec, state):
...
+
+
+@with_altair_and_later
+@spec_state_test
+@always_bls
+def test_invalid_signature_infinite_signature_with_participants(spec, state):
+    committee_indices = compute_committee_indices(spec, state, state.current_sync_committee)
+
+    block = build_empty_block_for_next_slot(spec, state)
+    # Exclude one participant whose signature was included.
```
Comment (protolambda) [inline tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py:138]: this comment is out of sync with implementation

File: tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py [hunk 2]
```diff
@@ -113,6 +113,36 @@ def test_invalid_signature_missing_participant(spec, state):
...
 
 
+@with_altair_and_later
+@spec_state_test
+@always_bls
+def test_invalid_signature_no_participants(spec, state):
+    committee_indices = compute_committee_indices(spec, state, state.current_sync_committee)
+
+    block = build_empty_block_for_next_slot(spec, state)
+    # Exclude one participant whose signature was included.
```
Comment (protolambda) [inline tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py:123]: Same with this comment

File: tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py [hunk 3]
```diff
@@ -113,6 +113,36 @@ def test_invalid_signature_missing_participant(spec, state):
...
+        sync_committee_bits=[False for _ in committee_indices],
+        sync_committee_signature=b'\x00' * 96
+    )
+    yield from run_sync_committee_processing(spec, state, block, expect_exception=True)
+
+
+@with_altair_and_later
+@spec_state_test
+@always_bls
+def test_invalid_signature_infinite_signature_with_participants(spec, state):
```
Comment (protolambda) [inline tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py:134]: probably also want a test with infinite signature, 0 participants, which should succeed.

**Review by protolambda — COMMENTED** [review]
Needs a few minor changes, I'll update this PR, looks good otherwise :+1:

File: tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py (hunk 3 above)
Comment (protolambda) [reply tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate.py:134]: Nvm, the success case is tested with `test_sync_committee_rewards_empty_participants` already

**Review by protolambda — APPROVED** [review]
LGTM
